[WIP] Lastin Egor

NotEgorL opened the pull request:
Created functions to find and separate feeds, using https://news.yahoo.com/rss/ like in the example.
Got into grips with the argparse module on a different computer(added parts of it as a docstring to the python script).

NotEgorL committed:
Add files via upload

Created functions to find and separate feeds, using https://news.yahoo.com/rss/ like in the example.
Got into grips with the argparse module on a different computer(added parts of it as a docstring to the python script).

NotEgorL commented:
Created functions to find and separate feeds, using https://news.yahoo.com/rss/ like in the example.
Got into grips with the argparse module on a different computer(added parts of it as a docstring to the python script).

AlexeiBuzuma commented on FinalTaskEgorLastin.py:
```diff
@@ -0,0 +1,86 @@
+
+# A list to hold all headlines,links,.....
+
+allheadlines=[]
```
в глобальной секции не должно быть никакой логики. нужно перенести в функции/классы.

AlexeiBuzuma commented on FinalTaskEgorLastin.py:
```diff
@@ -0,0 +1,86 @@
+    return links
+
+#def getTimes( rs_url)
+def getHeadlines( rss_url ):
```
есть много стилистических ошибок 
советую посмотреть в сторону утилиты `pycodestyle`

NotEgorL committed:
Add files via upload

NotEgorL commented:
mostly finished iteration 1 and had a go with iteration 2.
I added the files to the wrong place so this is my second try at adding to pull.

Iteration one has a working rss reader with a working limit and url parameter, Cleaned up the code a bit with logic and pycodestyle

NotEgorL committed:
Add files via upload

Added the option to add to file. the file is called news.txt and takes the standard feed from feedparser. the type of it is feedparserdict.
Still need to add classes and take ALL of the logic out, most of it has been done, same with pycodestyle

NotEgorL committed:
Add files via upload

pep'd and docstring'd. variable name changed

NotEgorL committed:
Add files via upload

added a -j for json dumping into the console. pycodestyled again

NotEgorL committed:
Add files via upload

Added more arguments. Namely a -date for task 3 and a -con for converting to  pdf and some other niche. The current functions are stubs for testing purposes.

AlexeiBuzuma commented on setup.py:
```diff
@@ -0,0 +1,8 @@
+from setuptools import setup, find_packages
+setup(
```
после установки утилиты в чистом докер контейнере через `pip install .` не появляется утилита rss-reader 
```bash
# rss-reader --help
bash: rss-reader: command not found
```

AlexeiBuzuma commented on k.py:
```diff
@@ -0,0 +1,90 @@
+"""import section"""
+
+version = 0.1
+parser = argparse.ArgumentParser()
```
все еще есть много кода, который хранится в глобальной секции 
нужно все перетащить как минимум в функции

AlexeiBuzuma commented on k.py:
```diff
@@ -0,0 +1,90 @@
+    print("a")
+
+
+def captureFeed(URL):
```
названия методом должны быть маленькими буквами через нижнее подчеркивание 
например `capture_feed`

AlexeiBuzuma commented on k.py:
```diff
@@ -0,0 +1,90 @@
+    """Outputs a json dump of feed.entries to a file that is called "news.txt".works """
+    open("news.txt", "w").close()
+    feed = captureFeed(args.URL)
+    f = open("news.txt", "w+")
```
для работы с файлами имеет смысл использовать контекстные менеджеры

NotEgorL committed:
Add files via upload
